solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -181,17 +191,28 @@ contract WalletRegistry is Ownable {
 
     /// @notice Requests a new wallet creation.
     /// @dev Can be called only by the owner of wallets.
-    ///      It starts a new DKG process.
+    ///      It locks the DKG and request a new random relay entry. It expects
+    ///      that the DKG process will be started once a new random relay entry
+    ///      gets generated.
     function requestNewWallet() external {
         require(msg.sender == walletOwner, "Caller is not the Wallet Owner");
 
         dkg.lockState();
-        // TODO: When integrating with the Random Beacon move `dkg.start` to a
-        // callback function. We need each DKG to be started with a unique
-        // and fresh relay entry.
-        dkg.start(
-            uint256(keccak256(abi.encodePacked(relayEntry, block.number)))
+
+        randomBeacon.requestRelayEntry(this);
+    }
+
+    /// @notice A callback that is executed once a new random relay entry gets
+    ///         generated. It starts the DKG process.
+    /// @dev Can be called only by the random beacon contract.
+    /// @param randomRelayEntry Random relay entry.
```

Comment:
Relay entry is always random, why not just `relayEntry`?

Reply:
9a5c5f1fc2d7886f35bf31629f89cc459c23e456